### PR TITLE
parse and report async gaps

### PR DESCRIPTION
**Pull request by yjbanov**
Related: https://github.com/flutter/flutter/issues/11479

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
Isn't crashy able to tell us if this fixes things? :)

**Comment by yjbanov** [comment]
This PR does not fix missing async stack traces. It only fixes missing async gaps, should they appear in the stack. I just confirmed using crashy that release builds are indeed missing async stack traces. Will comment in the original issue.

File: lib/src/stack_trace.dart [hunk 1]
```diff
@@ -4,7 +4,32 @@
...
+  final Chain chain = stackTrace is StackTrace
+      ? new Chain.forTrace(stackTrace)
+      : new Chain.parse(stackTrace);
+
+  final List<Map<String, dynamic>> frames = <Map<String, dynamic>>[];
+  for (int t = 0; t < chain.traces.length; t += 1) {
+    frames.addAll(chain.traces[t].frames.map(encodeStackTraceFrame));
+    if (t < chain.traces.length - 1) {
+      frames.add(asynchronousGapFrameJson);
+    }
```
Comment (cbracken) [inline lib/src/stack_trace.dart]: Drop the curly brackets for one-line `if`s.

File: lib/src/stack_trace.dart [hunk 2]
```diff
@@ -4,7 +4,32 @@
...
+///
+/// [stackTrace] must be [String] or [StackTrace].
+List<Map<String, dynamic>> encodeStackTrace(dynamic stackTrace) {
+  assert(stackTrace is String || stackTrace is StackTrace);
+  final Chain chain = stackTrace is StackTrace
+      ? new Chain.forTrace(stackTrace)
+      : new Chain.parse(stackTrace);
+
+  final List<Map<String, dynamic>> frames = <Map<String, dynamic>>[];
+  for (int t = 0; t < chain.traces.length; t += 1) {
```
Comment (cbracken) [inline lib/src/stack_trace.dart:23]: Why not `t++`?

**Review by cbracken — APPROVED** [review]

**Comment by cbracken** [comment]
lgtm

File: lib/src/stack_trace.dart (hunk 2 above)
Comment (yjbanov) [reply lib/src/stack_trace.dart:23]: That's Flutter style.

File: lib/src/stack_trace.dart (hunk 1 above)
Comment (yjbanov) [reply lib/src/stack_trace.dart]: Done.

**Pull request merged**